File: app/src/main/java/com/easycopy/screen/home/HomeNavigator.java
package com.easycopy.screen.home;

import android.net.Uri;

import androidx.documentfile.provider.DocumentFile;

import com.easycopy.screen.base.BaseNavigator;

/**
 * @author pankaj
 * @version 1.0
 * @since 2020-06-01
 */
public interface HomeNavigator extends BaseNavigator {
    void openFile(int requestCode);

    void openDirectory(int requestCode);

    DocumentFile getDocumentFile(Uri treeUri);
}

File: presentation/src/main/java/com/easycopy/screen/home/HomeNavigator.kt
package com.easycopy.screen.home

import android.net.Uri
import androidx.documentfile.provider.DocumentFile
import com.easycopy.screen.base.BaseNavigator

/**
 * @author pankaj
 * @version 1.0
 * @since 2020-06-01
 */
interface HomeNavigator : BaseNavigator {
    fun openFile(requestCode: Int)
    fun openDirectory(requestCode: Int)
    fun getDocumentFile(treeUri: Uri?): DocumentFile
}